kiosk/src/Components/MainMenu.tsx

```diff
@@ -4,35 +4,50 @@ import AddGameButton from "./AddGameButton";
 import GameList from "./GameList";
 import configData from "../config.json"
 import HighScoresList from "./HighScoresList";
+import { DeleteButton } from "./DeleteButton";
 import { tickEvent } from "../browserUtils";
+import DeletionModal from "./DeletionModal";
 
 interface IProps {
     kiosk: Kiosk
   }
 
 const MainMenu: React.FC<IProps> = ({ kiosk }) => {
-    const [buttonSelected, setButtonState] = useState(false);
+    const [addButtonSelected, setAddButtonState] = useState(false);
+    const [deleteButtonSelected, setDeleteButtonState] = useState(false);
+    const [deleteTriggered, setDeleteTriggered] = useState(false);
     const lockedClassName = kiosk.locked ? " locked" : "";
 
     const updateLoop = () => {
-        if (!buttonSelected && kiosk.gamepadManager.isUpPressed()) {
-            setButtonState(true);
+        if (!addButtonSelected && kiosk.gamepadManager.isUpPressed()) {
+            setAddButtonState(true);
         }
-        if (buttonSelected && kiosk.gamepadManager.isDownPressed()) {
-            setButtonState(false);
+        if (addButtonSelected && kiosk.gamepadManager.isDownPressed()) {
+            setAddButtonState(false);
         }
-        if (buttonSelected && kiosk.gamepadManager.isAButtonPressed()) {
+        if (!addButtonSelected && kiosk.selectedGame?.userAdded && kiosk.gamepadManager.isDownPressed()) {
+            setDeleteButtonState(true);
+        }
+        if (deleteButtonSelected && kiosk.gamepadManager.isUpPressed()) {
+            setAddButtonState(false);
+            setDeleteButtonState(false);
+        }
+        if (addButtonSelected && (kiosk.gamepadManager.isAButtonPressed() || kiosk.gamepadManager.isBButtonPressed())) {
             tickEvent("kiosk.addGamePageLoaded");
             kiosk.launchAddGame();
         }
+        if (deleteButtonSelected && (kiosk.gamepadManager.isAButtonPressed() || kiosk.gamepadManager.isBButtonPressed())) {
+            setDeleteTriggered(true);
+        }
     }
 
     useEffect(() => {
         if (!kiosk.locked) {
```

Comment:
It seems like this hook should list `kiosk` as a dependency, but maybe it never changes?

Reply:
Yeah, the instance of kiosk itself should never change. There are variables inside of that instance that change, but kiosk itself shouldn't, as far as I understand it.